pkg/mapper/iac-providers/arm/functions/variables.go

```diff
@@ -0,0 +1,51 @@
+/*
+    Copyright (C) 2020 Accurics, Inc.
+
+	Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+		http://www.apache.org/licenses/LICENSE-2.0
+
+	Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+*/
+
+package functions
+
+import (
+	"strings"
+
+	exp "github.com/VerbalExpressions/GoVerbalExpressions"
+	"go.uber.org/zap"
+)
+
+// Variables function runs variable against a regular
+// expression and return the variable key.
+//
+// For example:
+// if var = Variables('identityName'),
+// the function returns identityName as the key.
+func Variables(variable string) string {
+	const (
+		start = "variables('"
+		end   = "')"
+	)
+
+	key := strings.TrimPrefix(variable, "[")
+	key = strings.TrimRight(key, "]")
+	results := exp.New().
+		StartOfLine().Find(start).
+		BeginCapture().Anything().EndCapture().
+		Find(end).EndOfLine().
+		Captures(key)
```

Comment:
since this is common part in most functions, can we create a separate function for it?

Reply:
Should tackle better parameter resolution in the upcoming PR, we can discuss how to.